pytorch_blade/src/compiler/mlir/runtime/ral_context.cpp

```diff
@@ -197,11 +197,29 @@ torch::List<torch::Tensor> RalContext::CreateAndBindingOutputs(
     auto option = torch::device(dev_type)
                       .dtype(scalar_type)
                       .memory_format(torch::MemoryFormat::Contiguous);
-    torch::Tensor out_tensor = IsEmptyTensor(out_buf->shape())
-        ? torch::zeros(out_buf->shape(), option)
-        : torch::from_blob(
+    torch::Tensor out_tensor;
+    if (IsEmptyTensor(out_buf->shape())) {
+      out_tensor = torch::zeros(out_buf->shape(), option);
+    } else if (out_buf->owned()) {
+#ifdef TORCH_BLADE_BUILD_WITH_CUDA
+      auto deleter = (output_info.device == "cuda")
+          ? c10::cuda::CUDACachingAllocator::raw_delete
```

Comment:
Do we hardcode to use `CUDACachingAllocator` as the cuda allocator? I think ideally we should use PyTorch's `Allocator` interface (which has `raw_allocate` and `raw_deallocate` api).

Reply:
What is the difference between CUDACachingAllocator and Allocator?

Reply:
Allocator is the interface, CUDACachingAllocator is the default gpu implementation (users can implement their own cpu/gpu allocator)

Reply:
Thanks！ Done.